src/error.rs

```diff
@@ -1,26 +1,65 @@
+use thiserror::Error;
+
+use crate::proto::ConnectError;
+
+/// The set of observable errors when interacting with a Faktory server.
+#[derive(Debug, Error)]
+#[allow(clippy::manual_non_exhaustive)]
+pub enum Error {
+    /// The connection to the server, or one of its prerequisites, failed.
+    #[error("connection error: {0}")]
+    Connect(#[from] ConnectError),
+
+    /// Underlying io layer errors.
+    /// These are overwhelmingly network communication errors on the socket connection to the server.
+    #[error("underlying i/o: {0}")]
+    GenericIO(#[from] std::io::Error),
```

Comment:
I'm surprised this doesn't generate a warning. Rust tends to be pretty strict about only using capital letters for each word, which tends to get tripped up on things like IO. But if this doesn't generate a warning any more then :+1:

Reply:
Any particular reason to call this `GenericIO` as opposed to just `IO`? What makes it "generic"?

Reply:
No real reason, happy to change it back!